extensions/wikia/NjordPrototype/scripts/Mom.js

```diff
@@ -0,0 +1,303 @@
+		initMom = function () {
+			$momHeader.show();
+			$editButton.on('click', onEdit);
+			$saveButton.on('click', onSave);
+			$discardButton.on('click', onDiscard);
+			$deleteButton.on('click', function () {
+				$(this).parents('.mom-module').remove();
+			});
+			$(window).on('scroll', onScroll);
+		}, onEdit = function () {
+			$clonedMain = $mwContent.clone(true, true);
+			$('html, body').animate({
+				scrollTop: $momHeader.offset().top - 2
+			}, 600);
+			addEmpty();
+			$moms.addClass('mom-hidden');
+			$editMode.show();
+			$nonEditMode.hide();
+			$momBar.css('display', 'flex');
+			$momOverlays.show();
+			var options = {
+				cancel: '.mom-bar-content, .btn-group-right',
+				handle: '.mom-bar',
+				helper: 'clone',
+				items: '.mom-module-left, .mom-module-right',
+				placeholder: 'mom-add-module',
+				tolerance: 'pointer',
+				over: function (e, ui) {
+					var $child = $($(this).children('.mom-module')[0]);
+					if (ui.item[0] === $child[0]) {
+						//check second child if first (and only) child was grabbed
+						$child = $($(this).children('.mom-module')[1]);
+					}
+					if ($child.hasClass('mom-add-module')) {
+						ui.placeholder.insertBefore($child);
+					}
+				}
+			};
+			options['connectWith'] = '.rcs-container';
+			$leftColumn.sortable(options);
+			options['connectWith'] = '.lcs-container';
+			$rightColumn.sortable(options);
+			$leftColumn.sortable('enable');
+			$rightColumn.sortable('enable');
+			$momBarContent.each(function () {
+				$(this).attr('contenteditable', true);
+			});
+		}, onSave = function () {
+			removeEmpty();
+			$editMode.hide();
+			$nonEditMode.show();
+			$mainContent.startThrobbing();
+			$moms.removeClass('mom-hidden');
+			$moms.removeClass('mom-edit');
+			$momBar.css('display', 'none');
+			$momOverlays.hide();
+			$leftColumn.sortable('disable');
+			$rightColumn.sortable('disable');
+			$momBarContent.each(function () {
+				$(this).removeAttr('contenteditable');
+			});
+			$('.mom-edit-btn').show();
+			$('.mom-delete-btn').show();
+			$('.mom-content').show();
+			$('.mom-save-btn').hide();
+			$('.mom-discard-btn').hide();
+			$('.mom-wiki-markup').hide();
+			var left = [],
+				right = [];
+			$('.lcs-container .mom-module').each(function () {
+				var data = $(this).data();
+				if (typeof data.title != 'undefined') {
+					left.push(JSON.stringify({
+						'title': data.title,
+						'text': $(this).find('.mom-bar .mom-bar-content').text()
+					}));
+				}
+			});
+			$('.rcs-container .mom-module').each(function () {
+				var data = $(this).data();
+				if (typeof data.title != 'undefined') {
+					right.push(JSON.stringify({
+						'title': data.title,
+						'text': $(this).find('.mom-bar-content').text()
+					}));
+				}
+			});
+			$.nirvana.sendRequest({
+				controller: 'NjordController',
+				method: 'reorder',
+				type: 'POST',
+				data: {
+					'page': wgTitle,
+					'left': left,
+					'right': right
+				},
+				callback: onDataSaved,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the declaration of "onDataSaved" before this usage. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)